src/core/utils/parse_cmdline.cpp

```diff
@@ -43,28 +45,30 @@ static bool IsSpaceOrNewLine(char aChar)
     return (aChar == ' ') || (aChar == '\t') || (aChar == '\r') || (aChar == '\n');
 }
 
+static bool IsEscapable(char aChar)
+{
+    return IsSpaceOrNewLine(aChar) || (aChar == '\\');
+}
+
 otError CmdLineParser::ParseCmd(char *aString, uint8_t &aArgc, char *aArgv[], uint8_t aArgcMax)
 {
     otError error = OT_ERROR_NONE;
     char *  cmd;
 
     aArgc = 0;
 
-    for (cmd = aString; IsSpaceOrNewLine(*cmd) && *cmd; cmd++)
-        ;
-
-    if (*cmd)
+    for (cmd = aString; *cmd; cmd++)
     {
-        aArgv[aArgc++] = cmd++; // the first argument
-    }
-
-    for (; *cmd; cmd++)
-    {
-        if (IsSpaceOrNewLine(*cmd))
+        if ((*cmd == '\\') && IsEscapable(*(cmd + 1)))
+        {
+            memmove(cmd, cmd + 1, strlen(cmd));
```

Comment:
nit: `memmove(cmd, cmd + 1, strlen(cmd+1) + 1)` would be clearer to us. We got two reviewers thinking this is wrong at first glance :)

Reply:
@wgtdkp :)

Well, yes, it would be clearer but take more program memory as well. Maybe it's better to just add a comment clarifying that we want to move everything from the next character until the null terminator, both included?

Reply:
it works to me.

Reply:
Done.